### ISSUE TITLE
suggestion: # nolint like exclussion for path problems

### ISSUE THREAD

**Issue by ThierryO** [r-lib/lintr#899]
I'm getting lots of false positive when using the portable file paths linter.

- string of a regex pattern: e.g. `"\\.jpg$"`
- string with git refs. e.g. `"origin/main"`, `"refs/heads/branch"`
- constructing long URL with paste to stay belong the long line limit.

```
paste(
  "https://github.com/r-lib/lintr/blob/master/.github",
  "workflows/lint-changed-files.yaml", sep = "/"
)
```

The current solutions are IMHO subpar:
- turning the portable file paths linter off. Then we no longer check for portable file paths.
- replacing all the false positive with `file.path(..., fsep = "/")` or `paste(..., sep = "/")`. This make the code much less readable.
- adding `# nolint` to the false negative. This turns off all checking.

A more elegant solution could be something like adding `# nolintpath` which only excludes the path related linters.

**Comment by AshesITR** [comment]
Thanks für your suggestion.
Something like this is already possible in the current development version using, e.g.,
`#nolint: absolute_path_linter.`.

Please let us know if you encounter any issues with this new feature.

**Comment by ThierryO** [comment]
Thanks for info. It works like a charm.

**Comment by ThierryO** [comment]
I spoke too soon. When I exclude a different linter than the one causing the other, the line is ignored.
An example to clarify my problem. `lint_package()` reports a line as failing the `nonportable_path_linter`. I add
add `#nolint: assignment_linter` to a line (not the linter while is failing) . Now `lint_package()` no longer reports the line at all. While the linter still existst.

Note that I am using `options(lintr.linter_file = system.file("lintr", package = "checklist"))` with this [lintr file](https://github.com/inbo/checklist/blob/zenodo/inst/lintr). Below the session info.

```
─ Session info ─────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.2 (2021-11-01)
 os       Ubuntu 20.04.3 LTS
 system   x86_64, linux-gnu
 ui       RStudio
 language nl_BE:nl
 collate  nl_BE.UTF-8
 ctype    nl_BE.UTF-8
 tz       Europe/Brussels
 date     2022-01-12
 rstudio  2022.01.1-daily+167 Prairie Trillium (desktop)
 pandoc   2.14.2 @ /usr/lib/rstudio/bin/quarto/bin/ (via rmarkdown)

─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package      * version    date (UTC) lib source
 askpass        1.1        2019-01-13 [1] CRAN (R 4.1.0)
 assertthat     0.2.1      2019-03-21 [1] CRAN (R 4.1.0)
 bit            4.0.4      2020-08-04 [1] CRAN (R 4.1.0)
 bit64          4.0.5      2020-08-30 [1] CRAN (R 4.1.0)
 blob           1.2.2      2021-07-23 [1] CRAN (R 4.1.1)
 cachem         1.0.6      2021-08-19 [1] CRAN (R 4.1.1)
 callr          3.7.0      2021-04-20 [1] CRAN (R 4.1.0)
 checklist      0.2.3      2022-01-12 [1] local
 cli            3.1.0      2021-10-27 [1] CRAN (R 4.1.2)
 codemetar      0.3.3      2022-01-04 [1] CRAN (R 4.1.2)
 codetools      0.2-18     2020-11-04 [1] CRAN (R 4.1.0)
 crayon         1.4.2      2021-10-29 [1] CRAN (R 4.1.2)
 credentials    1.3.2      2021-11-29 [1] CRAN (R 4.1.2)
 crul           1.2.1.99   2022-01-05 [1] https://ropensci.r-universe.dev (R 4.1.2)
 curl           4.3.2      2021-06-23 [1] CRAN (R 4.1.0)
 cyclocomp      1.1.0      2016-09-10 [1] CRAN (R 4.1.0)
 DBI            1.1.2      2021-12-20 [1] CRAN (R 4.1.2)
 desc           1.4.0.9000 2022-01-05 [1] https://ropensci.r-universe.dev (R 4.1.2)
 devtools       2.4.3      2021-11-30 [1] CRAN (R 4.1.2)
 digest         0.6.29     2021-12-01 [1] CRAN (R 4.1.2)
 dplyr          1.0.7      2021-06-18 [1] CRAN (R 4.1.0)
 ellipsis       0.3.2      2021-04-29 [1] CRAN (R 4.1.0)
 evaluate       0.14       2019-05-28 [1] CRAN (R 4.1.0)
 fansi          0.5.0      2021-05-25 [1] CRAN (R 4.1.0)
 fastmap        1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
 fauxpas        0.5.0      2020-04-13 [1] CRAN (R 4.1.0)
 fortunes       1.5-4      2016-12-29 [1] CRAN (R 4.1.0)
 fs             1.5.2      2021-12-08 [1] CRAN (R 4.1.2)
 generics       0.1.1      2021-10-25 [1] CRAN (R 4.1.1)
 gert           1.5.0      2022-01-03 [1] CRAN (R 4.1.2)
 git2r          0.29.0     2021-11-22 [1] CRAN (R 4.1.2)
 glue           1.6.0      2021-12-17 [1] CRAN (R 4.1.2)
 hms            1.1.1      2021-09-26 [1] CRAN (R 4.1.1)
 htmltools      0.5.2      2021-08-25 [1] CRAN (R 4.1.1)
 httpcode       0.3.0      2020-04-10 [1] CRAN (R 4.1.0)
 httr           1.4.2      2020-07-20 [1] CRAN (R 4.1.0)
 jsonlite       1.7.2      2020-12-09 [1] CRAN (R 4.1.0)
 knitr          1.37       2021-12-16 [1] CRAN (R 4.1.2)
 lazyeval       0.2.2      2019-03-15 [1] CRAN (R 4.1.0)
 lifecycle      1.0.1      2021-09-24 [1] CRAN (R 4.1.1)
 lintr          2.0.1.9000 2022-01-12 [1] Github (r-lib/lintr@6e0d152)
 lubridate      1.8.0      2021-10-07 [1] CRAN (R 4.1.1)
 magrittr       2.0.1      2020-11-17 [1] CRAN (R 4.1.0)
 memoise        2.0.1      2021-11-26 [1] CRAN (R 4.1.2)
 n2khelper      0.5.0      2022-01-12 [1] local
 odbc           1.3.3      2021-11-30 [1] CRAN (R 4.1.2)
 openssl        1.4.6      2021-12-19 [1] CRAN (R 4.1.2)
 pillar         1.6.4      2021-10-18 [1] CRAN (R 4.1.1)
 pingr          2.0.1      2020-06-22 [1] CRAN (R 4.1.0)
 pkgbuild       1.3.1      2021-12-20 [1] CRAN (R 4.1.2)
 pkgconfig      2.0.3      2019-09-22 [1] CRAN (R 4.1.0)
 pkgdown        2.0.1      2021-12-03 [1] CRAN (R 4.1.2)
 pkgload        1.2.4      2021-11-30 [1] CRAN (R 4.1.2)
 plyr           1.8.6      2020-03-03 [1] CRAN (R 4.1.0)
 prettyunits    1.1.1      2020-01-24 [1] CRAN (R 4.1.0)
 processx       3.5.2      2021-04-30 [1] CRAN (R 4.1.0)
 ps             1.6.0      2021-02-28 [1] CRAN (R 4.1.0)
 purrr          0.3.4      2020-04-17 [1] CRAN (R 4.1.0)
 R6             2.5.1      2021-08-19 [1] CRAN (R 4.1.1)
 rcmdcheck      1.4.0      2021-09-27 [1] CRAN (R 4.1.1)
 Rcpp           1.0.7      2021-07-07 [1] CRAN (R 4.1.0)
 remotes        2.4.2      2021-11-30 [1] CRAN (R 4.1.2)
 rex            1.2.1      2021-11-26 [1] CRAN (R 4.1.2)
 rlang          0.4.12     2021-10-18 [1] CRAN (R 4.1.1)
 rmarkdown      2.11       2021-09-14 [1] CRAN (R 4.1.1)
 RODBC          1.3-19     2021-09-16 [1] CRAN (R 4.1.1)
 rorcid         0.7.1.91   2021-09-21 [1] https://ropensci.r-universe.dev (R 4.1.1)
 RPostgreSQL    0.7-3      2021-10-22 [1] CRAN (R 4.1.1)
 rprojroot      2.0.2      2020-11-15 [1] CRAN (R 4.1.0)
 rstudioapi     0.13       2020-11-12 [1] CRAN (R 4.1.0)
 sessioninfo    1.2.2      2021-12-06 [1] CRAN (R 4.1.2)
 stringi        1.7.6      2021-11-29 [1] CRAN (R 4.1.2)
 stringr        1.4.0      2019-02-10 [1] CRAN (R 4.1.0)
 sys            3.4        2020-07-23 [1] CRAN (R 4.1.0)
 testthat       3.1.1      2021-12-03 [1] CRAN (R 4.1.2)
 tibble         3.1.6      2021-11-07 [1] CRAN (R 4.1.2)
 tidyr          1.1.4      2021-09-27 [1] CRAN (R 4.1.1)
 tidyselect     1.1.1      2021-04-30 [1] CRAN (R 4.1.0)
 usethis        2.1.5      2021-12-09 [1] CRAN (R 4.1.2)
 utf8           1.2.2      2021-07-24 [1] CRAN (R 4.1.1)
 vctrs          0.3.8      2021-04-29 [1] CRAN (R 4.1.0)
 whisker        0.4        2019-08-28 [1] CRAN (R 4.1.0)
 withr          2.4.3      2021-11-30 [1] CRAN (R 4.1.2)
 xfun           0.29       2021-12-14 [1] CRAN (R 4.1.2)
 xml2           1.3.3      2021-11-30 [1] CRAN (R 4.1.2)
 xmlparsedata   1.0.5      2021-03-06 [1] CRAN (R 4.1.0)
 xopen          1.0.0      2018-09-17 [1] CRAN (R 4.1.0)
 yaml           2.2.1      2020-02-01 [1] CRAN (R 4.1.0)

 [1] /home/thierry/R/x86_64-pc-linux-gnu-library/4.0
 [2] /usr/local/lib/R/site-library
 [3] /usr/lib/R/site-library
 [4] /usr/lib/R/library
```

**Comment by AshesITR** [comment]
You need a dot at the end of the exclusion list, please try `#nolint: assignment_linter.`.

**Comment by ThierryO** [comment]
I see. Now I get it working as expected.
